test/test-parse-function.cpp

```diff
@@ -851,6 +851,37 @@ TEST(test_parse, unnamed_function_expression_without_body) {
   }
 }
 
+TEST(test_parse, arrow_function_invoked_with_parens) {
+  {
+    padded_string code(u8"(() => {})()"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    p.parse_and_visit_module(v);
+    EXPECT_THAT(v.visits, ElementsAre("visit_enter_function_scope",
+                                      "visit_enter_function_scope_body",
+                                      "visit_exit_function_scope",
+                                      "visit_end_of_module"));
+    EXPECT_EQ(v.errors.size(), 0);
+  }
+}
+
+TEST(test_parse, arrow_function_invoked_no_parens) {
+  {
+    padded_string code(u8"() => {}()"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    p.parse_and_visit_module(v);
+    EXPECT_THAT(v.visits, ElementsAre("visit_enter_function_scope",
+                                      "visit_enter_function_scope_body",
+                                      "visit_exit_function_scope",
+                                      "visit_end_of_module"));
+
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_OFFSETS(
+                              &code, error_missing_parens_self_invoked_function,
+                              func, 0, u8"() => {}")));
+  }
+}
```

Comment:
Nit: I'd prefer these tests to be `parse_expression` tests in `test/test-parse-expression.cpp`. `parse_expression` tests can be a bit shorter. But what you wrote is good too.

Reply:
I didn't make a change for this - let me know if this is a blocker for you